marklogic-unit-test-modules/src/main/ml-modules/root/test/test-controller.xqy

```diff
@@ -280,11 +281,14 @@ declare function format-junit($result as element())
 			element testcase {
 				attribute classname {fn:data($test/@name)},
 				attribute name {fn:data($test/@name)},
+        attribute tests {fn:count($test/test:result)},
+        attribute success {fn:count($test/test:result[@type = 'success'])},
+        attribute failed {fn:count($test/test:result[@type = 'fail'])},
 				attribute time {fn:data($test/@time)},
-				for $result in ($test/test:result)[1]
-				where $result/@type = "fail"
+				for $result at $idx in $test/test:result[@type = "fail"]
```

Comment:
Is this a bug fix here? The code was previously only looking at the first test result, which may not have failed, but the second or later result may have failed?

Reply:
Also, I'm not sure if this is valid JUnit legacy XML. Doing some googling to see if multiple `failure` elements are allowed within a `testcase` element.